### PR TITLE
Handling of flicks' icons' overlays' changes

### PR DESCRIPTION
**Pull request by AlexMorgan3817**
## Changelog
:cl:
refactor: Flicks now handling by /datum/proc/flicker
refactor: Icon, icon state now should handling by /atom/proc/SetIcon and SetIconState.
refactor: Overlays operations refactored to procs as cut_overlays, add_overlays, associate and etc.
/:cl:

### CODE REVIEW

**Comment by AlexMorgan3817** [comment]
:weary:

**Comment by AlexMorgan3817** [comment]
Ready for review

File: code/__HELPERS/icons.dm [hunk 1]
```diff
@@ -1038,4 +1038,47 @@ proc/get_average_color(var/icon, var/icon_state, var/image_dir)
...
+		overlays = _overlays //No need to .Copy, byond copy ovelays' list by it self
+	if(isByEvent)
+		GLOB.flicker_event.register(D, src, .proc/icon_synchronization)
+
+/image/proc/SyncWithAtom(atom/D)
+	if(istype(D))
+		GLOB.flicker_event.register(D, src, .proc/flick_synchronization)
+		GLOB.update_icon_event.register(D, src, .proc/icon_synchronization)
+	else
+		CRASH("[D](\ref[D]) is not datum, aborting /image/proc/SyncWithDatum. Additional info: {[json_encode(args)]}")
```
Comment (AlexMorgan3817) [inline code/__HELPERS/icons.dm:1079]:
```suggestion
		CRASH("[D](\ref[D]) is not datum, aborting /image/proc/SyncWithDatum. Additional info: {[json_encode(args)]}")
```
```suggestion
		CRASH("[D](\ref[D]) is not atom, aborting /image/proc/SyncWithDatum. Additional info: {[json_encode(args)]}")
```

**Comment by AlexMorgan3817** [comment]
Okey, now time for icons...

**Comment by AlexMorgan3817** [comment]
Done, i think

**Review by TheShown911 — COMMENTED** [review]
That's a lot of damage

**Comment by AlexMorgan3817** [comment]
There is no exploits hidden under 3.8k of changed lines, trust me.

**Review by mattroks101 — APPROVED** [review]
Don't see any obvious errors. Someone else can look over it as well though.

**Review by TorinoFermic — APPROVED** [review]
Nothing else jumps at me.

**Pull request merged**